src/main/webapp/script.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Event processing for upload button
+ */
+function fileUploadDisplay() {
+  document.querySelector('.bg-model').style.display = 'flex';
+}
+
+/**
+ * Event processing for close button
+ */
+function closeDiv() {
+  document.querySelector('.bg-model').style.display = 'none';
+}
+
+/**
+ * Makes the form for uploading an image visible and adds the 'action' to it by fetching
+ * the url(that the form needs to post to) from the servlet
+ */
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload-url')
+      .then((response) => response.text())
+      .then((imageUploadUrl) => {
+        const uploadForm = document.getElementById('upload-barcode-form');
+        uploadForm.action = imageUploadUrl;
+        uploadForm.classList.remove('hidden');
```

Comment:
I'm not sure why you need this. Isn't the element nested inside a hidden element at the beginning?

Reply:
Ok, so, the form has a dependency on blobstore  and unless the blobstore provides it with an action url, our form will be incomplete. So, it is hidden till it gets the blobstore url.

To avoid errors.
As the upload image div could be opened but if the form doesn't have a url yet and user tries to submit something then it'll not be a good thing.

Reply:
So, do I understand this correctly: If the blobstore URL is not yet present when I click the upload image button, I'm seeing an empty upload form?

This seems a bit odd. Maybe in a future change you could replace this by a loading spinner or something like that.

Reply:
Yes it is, at the moment, supposed to show an empty form if the blobstore url is is not present.
I'll add this point in the ToDo list of the project and will check out the spinner you mentioned.